test/phoenix_component/verify_test.exs

```diff
@@ -964,6 +964,22 @@ defmodule Phoenix.ComponentVerifyTest do
            """
   end
 
+  test "global includes" do
+    warnings =
+      capture_io(:stderr, fn ->
+        defmodule GlobalIncludes do
+          use Phoenix.Component
+
+          attr :id, :any, required: true
+          attr :rest, :global, include: ~w(form)
+          def button(assigns), do: ~H|<button id={@id} {@rest}>button</button>|
+          def any_render(assigns), do: ~H|<.button id="123" form="my-form" />|
+        end
+     end)
+
+    assert warnings == ""
```

Comment:
This runs concurrently, which means you can capture stderr from other tests. In this case, I would remove `capture_io`. You can also add an assertion that the includes are rendered correctly. :)

Reply:
We need to test the warnings and we do the cpature_io above in another test, so I think `async: false` is the best choice?

Reply:
In this case there is no warning. So it should be fine to not have `capture_io`. Then we can enable `mix test --warnings-as-errors` on CI such that, if it does warn and we don't capture it, then it will fail the suite anyway.